File: server/app/models/search_config.py
"""
Search engine configuration model.

Stores a single search engine configuration (id=1).
When is_active is False or api_key is empty, the search tool is not available.
"""

from sqlalchemy import Column, Integer, String, DateTime, Text, Boolean
from datetime import datetime
from app.database import Base
from .base import LOCALTIME


class SearchConfig(Base):
    """
    Search engine configuration model.

    Only one record exists (id=1). When is_active is False or api_key is empty,
    the WebSearchTool will not be added to the agent's tool list.
    """

    __tablename__ = "search_config"

    id = Column(Integer, primary_key=True, default=1)
    provider = Column(String(50), nullable=False, default='tavily')
    api_key = Column(String(255), nullable=False, default='')
    description = Column(Text, nullable=False, default='')
    is_active = Column(Boolean, nullable=False, default=False)
    updated_at = Column(DateTime(timezone=True), default=datetime.now, server_default=LOCALTIME, onupdate=datetime.now, nullable=False)

    def is_available(self) -> bool:
        """
        Check if the search engine is available for use.

        Returns:
            True if is_active is True and api_key is not empty.
        """
        return self.is_active and bool(self.api_key)

File: server/internal/model/search_config.go
package model

import "time"

// SearchConfig stores the web search provider configuration.
// Currently supports Tavily as the search provider.
type SearchConfig struct {
	ID          int64     `gorm:"primaryKey;autoIncrement;type:INTEGER PRIMARY KEY AUTOINCREMENT;default:1" json:"id"`
	Provider    string    `gorm:"type:varchar(50);not null;default:'tavily'" json:"provider"`
	APIKey      string    `gorm:"type:varchar(255);not null;default:'';column:api_key" json:"api_key"`
	Description string    `gorm:"type:text;not null;default:''" json:"description"`
	IsActive    bool      `gorm:"not null;default:false;column:is_active" json:"is_active"`
	UpdatedAt   time.Time `gorm:"not null;autoUpdateTime" json:"updated_at"`
}

func (SearchConfig) TableName() string { return "search_config" }

// IsAvailable returns true if the search provider is active and has a valid API key.
func (sc *SearchConfig) IsAvailable() bool {
	return sc.IsActive && sc.APIKey != ""
}
